CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+# CONTRIBUTING
+
+Thank you for considering contributing to the Outrigger CLI!
+
+## Quality Contributions
+
+* Make sure your branch will compile.
+* Make sure your branch passes our static analysis checks.
+* Make sure your branch conforms with go fmt standards.
+* Manually test your changes.
+
+## User Interactions
+
+One of the key goals of this project is to promote a positive developer
+experience. Every interaction should be thought of with the following points:
+
+* Are you providing the user with enough context about what's being asked or being done?
+* Does the user expect to wait? Might the user think the tool stalled?
+* Is there black box business happening that could be made more transparent?
+
+We have a slightly complex logging API to support addressing these concerns.
+(See ./util/logging.go)
+
+Here are a few conventions:
+
+* **Starting a task that could take more than 5 seconds:**
+  * `cmd.out.Spin("Preparing the sauce")`
+* **Use the correct method to log operational results: (Pick one)**
+  * `cmd.out.Success("Sauce is Ready.")`
+  * `cmd.out.Warn("Sauce is burnt on the bottom.")`
+  * `cmd.out.Oops("Discard this sauce and try again.")`
```

Comment:
Oops is temporary, there are a number of "refactoring in progress" things going on before all the specific API details are right.